url5a01f2c1d9_core9_core_tgz-pJ8-io_core9_core_boot_BootstrapFrameworkJ8/src/net/xeoh/plugins/base/impl/classpath/loader/HTTPLoader.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * InternalClasspathLoader.java
+ * 
+ * Copyright (c) 2009, Ralf Biedert All rights reserved.
+ * 
+ * Redistribution and use in source and binary forms, with or without modification, are
+ * permitted provided that the following conditions are met:
+ * 
+ * Redistributions of source code must retain the above copyright notice, this list of
+ * conditions and the following disclaimer. Redistributions in binary form must reproduce the
+ * above copyright notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ * 
+ * Neither the name of the author nor the names of its contributors may be used to endorse or
+ * promote products derived from this software without specific prior written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * 
+ */
+package net.xeoh.plugins.base.impl.classpath.loader;
+
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+
+import net.xeoh.plugins.base.impl.PluginManagerImpl;
+
+/**
+ * @author rb
+ *
+ */
+public class HTTPLoader extends FileLoader {
+
+    /**
+     * @param pluginManager
+     */
+    public HTTPLoader(PluginManagerImpl pluginManager) {
+        super(pluginManager);
+    }
+
+    /* (non-Javadoc)
+     * @see net.xeoh.plugins.base.impl.loader.AbstractLoader#handlesURI(java.net.URI)
+     */
+    @Override
+    public boolean handlesURI(URI uri) {
+        if (uri.getScheme().equals("http")) return true;
+
+        return false;
+    }
+
+    /* (non-Javadoc)
+     * @see net.xeoh.plugins.base.impl.loader.AbstractLoader#loadFrom(java.net.URI)
+     */
+    @Override
+    public void loadFrom(URI url) {
+        // Handle http files
+        if (url.getScheme().equals("http")) {
+
+            try {
+                // download the file (TODO: download could be improved a bit ...)
+                java.io.File tmpFile = java.io.File.createTempFile("jspfplugindownload", ".jar");
+                final FileOutputStream fos = new FileOutputStream(tmpFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fos*. The resource is referenced at line 80. The resource is closed at line 83. There are other execution paths that do not contain closure statements, for example, when URI.toURL throws an exception. Ensure *fos* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).